packages/functional-tests/tests/react-conversion/resetPassword.spec.ts

```diff
@@ -56,52 +50,186 @@ test.describe('reset password', () => {
+    await diffPage.waitForURL(
+      `${target.contentServerUrl}/reset_password_verified`
+    );
+
+    // Wait for initial page to automatically redirect once password is reset
+    await page.waitForURL(`${target.contentServerUrl}/signin`);
+
+    // Verify password reset confirmation page is rendered
+    expect(await resetPassword.resetPasswordConfirmedReact(diffPage)).toBe(
+      true
+    );
+
+    await diffPage.close();
+
+    // Verify initial page redirected to sign in and sign in page rendered
+    expect(await login.emailHeader.isEnabled()).toBe(true);
+
+    await login.setEmail(credentials.email);
+    await login.clickSubmit();
+    await page.waitForURL(`${target.contentServerUrl}/signin`);
+
+    await login.setPassword(NEW_PASSWORD);
+    await login.clickSubmit();
+    await page.waitForURL(`${target.contentServerUrl}/settings`);
+
     expect(
-      await diffPage
-        .locator('text="Remember your password? Sign in"')
+      await page
+        .getByRole('heading', { name: 'Settings', level: 2 })
         .isEnabled()
-    ).toBeTruthy();
+    ).toBe(true);
+
+    // Cleanup requires setting this value to correct password
+    credentials.password = NEW_PASSWORD;
+  });
 
-    await diffPage.locator('input[name="newPassword"]').fill(NEW_PASSWORD);
-    await diffPage.locator('input[name="confirmPassword"]').fill(NEW_PASSWORD);
+  /* TODO test reset pw validation errors
+   * - short password (e.g., "pass")
+   * - common password (e.g., "password")
+   * - user emails (e.g., credentials.email)
+   *
+   * In these cases, confirm that error feedback is displayed and form is not submitted
+   */
+
+  test('visit confirmation screen without initiating reset_password, user is redirected to /reset_password', async ({
+    target,
+    page,
+    pages: { resetPassword },
+  }) => {
+    await page.goto(getReactFeatureFlagUrl(target, '/confirm_reset_password'));
+
+    // Verify its redirected to react reset password page
+    expect(await page.locator('#root').isEnabled()).toBe(true);
+    expect(await resetPassword.beginResetPasswordHeadingReact()).toBe(true);
+  });
 
-    const pageWaitForNavigation = page.waitForNavigation();
-    const diffPageWaitForNavigation = diffPage.waitForNavigation();
-    await diffPage.locator('text="Reset password"').click();
-    await diffPageWaitForNavigation;
-    await pageWaitForNavigation;
+  test('open /reset_password page from /signin', async ({
+    credentials,
+    page,
+    pages: { login },
+  }) => {
+    await login.goto();
+    await login.setEmail(credentials.email);
+    await login.submit();
+    await login.clickForgotPassword();
+    // Verify react page has been loaded - to be enabled when link to react page from sign in is active
+    // await page.waitForSelector('#root');
+  });
 
-    expect(
-      await diffPage.locator('text="Your password has been reset"').isEnabled()
-    ).toBeTruthy();
+  test('enter an email with leading whitespace', async ({
```

Comment:
FWIW, I think `enter an email with leading whitespace`and `enter an email with trailing whitespace` should be unit tests.

Reply:
I was wondering about that as I was porting over existing functional tests - feels like there are quite a few grey area where our functional tests could be moved to unit tests. E.g., some spots where we were checking if all page elements were rendered. 

That's part of the reason why I've marked the PR as a draft - I want to circle back on questions like this about what we need to be testing in functional tests for the reset password flow.